v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/BinaryAvroDeserializer.java

```diff
@@ -35,7 +35,7 @@ public BinaryAvroDeserializer(Schema schema) {
     this.schema = schema;
   }
 
-  public GenericRecord deserialize(String topic, Headers header, byte[] bytes) {
+  public Object deserialize(String topic, Headers header, byte[] bytes) {
```

Comment:
can we typecast in this method instead and return GenericRecord?

Reply:
We need to use `Object` to override parent methods.